### PR TITLE
Music and Sound Effects

### PR DESCRIPTION
**Pull request by PBWiseman**
## Describe your changes

This branch will be adding appropriate, and free, ambient music and sound effects for the game

## Issue number and link

#111 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have written any applicable tests
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] This complies with the [product coding guidelines](https://github.com/BIT-Studio-3/Space-Rescue/wiki/Code-Review-Format)

### CODE REVIEW

**Comment by PBWiseman** [comment]
Status update: Music is functioning, and 4 tracks have been added to the game. Sound system is added as well as volume controls.

**Comment by PBWiseman** [comment]
I've added sound effects for button clicking, button hovering, going too close to the black hole, and boosting. Can I get some feedback on the sound effects, the music, the new menu tab, and the coding. Also, suggestions for what else you want sound effects for.

File: Assets/Scripts/Managers/SFXManager.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        DontDestroyOnLoad(this.gameObject);
+        audioSource = GetComponent<AudioSource>();
+        SetVolume();
+        if(SceneManager.GetSceneByName("Settings").isLoaded)
+        { //gets the settings scene and checks if its loaded
+            isSettings = true;
+        }
+        else if(SceneManager.GetSceneByName("Planet").isLoaded)
+        { //gets the settings scene and checks if its loaded
```
Comment (ChaseBENNC9) [inline Assets/Scripts/Managers/SFXManager.cs:25]: I assume this comment should say planet instead of Settings

**Review by ChaseBENNC9 — COMMENTED** [review]
Really like all sound effects you've chosen.  No issues with code from what I looked at. 
not you if you looked into it but my suggestion would be to find a sound for collecting animals while on the planet.

File: Assets/Scripts/Managers/SFXManager.cs (hunk 1 above)
Comment (PBWiseman) [reply Assets/Scripts/Managers/SFXManager.cs:25]: Yup. Fixed.

**Review by 3MKJOE58 — COMMENTED** [review]
Like the sound effects, as Chase suggested for picking the animals whenever an animal is picked maybe you can play that animal sound.

**Review by erokasyuart — CHANGES_REQUESTED** [review]
I really like the title screen music, very ominous, and the atmosphere of the in-game music is very exploratory and adventurous.

In my **_personal_** opinion, it feels a little odd to have the in-game music playing over the pause menu. I'm not suggesting removing the music entirely, I just think there should be a slight audial difference when the pause menu is open-- like reducing the sound slightly so the player can focus better when they're reading the settings and etc.

I also think it would be good to have some kind of sound effect for when the boosters are recharging. In my testing, there's an ascending synth sound which I thought was the sound effect for the boosters recharging until I realised it was actually part of the music.

I'm also conflicted about how I feel about the button select sound only being able to play once the last one had fully ended.

I think that the title screen buttons should have a separate sound that differentiates itself to the settings button, maybe with a softer, rounder sound.

I'm also conflicted about the music on the planet, I can tell its supposed to be carried over but since it's the same audio, it gets chopped for a second while the scene loads. Would also be good to have some sounds emitting from the animals! But maybe wait until Chase's animal models are added in first...

**Comment by PBWiseman** [comment]
TODO list:
[ ] Planet landing
[ ] Planet takeoff
[ ] Collecting animals
[ ] Work out pause menu changes? Maybe a separate sound slider for menu music?
[ ] Booster recharge sound effect
[ ] Animal noises (Probably waiting until animals are implemented? I can get a list and have the sounds ready though)
[x] Rounder title screen button noises
[x] Fix button select noise bug

**Comment by PBWiseman** [comment]
I have decided to not implement a booster recharge sound as the boosters are recharging pretty much constantly and whatever sound was added would get very old very quickly.

**Comment by PBWiseman** [comment]
Current status of branch:  
- Planet landing and takeoff sounds are in the files. There isn't currently any loading screen or animation for this so there is nowhere to put it. Will add them to the game when there is.
- No animal sound effects as I am waiting to see the full list of animals that end up being implemented. Will be another sprint
- No pickup sound as I am waiting to confirm what animation we will have for this and see what sound would work best. Will be in another branch.

Everything else I was aiming for has been finished. Branch is ready to be reviewed.

**Review by ChaseBENNC9 — APPROVED** [review]
### Code:  ✅
Great use of comments in the scripts, they make the code easy to understand and explain the purpose when needed.
### Functionality ✅
All the sound effects seem to be working with no issues. game has been tested fully in the editor and in a build with no issues


### Tests ✅
none 
#### Bugs
no bugs were found in test

**Review by erokasyuart — APPROVED** [review]
My issues have been resolved and this has been greatly improved!

**Review by 3MKJOE58 — APPROVED** [review]
Works good, approved for merge

**Pull request merged**